src/libasr/codegen/asr_to_c.cpp

```diff
@@ -490,6 +490,20 @@ R"(
         array_types_decls += open_struct + body + end_struct;
     }
 
+    void visit_ComplexConstant(const ASR::ComplexConstant_t &x) {
+        headers.insert("complex");
+
+        std::string re = std::to_string(x.m_re);
+        std::string im = std::to_string(x.m_im);
+
+        std::string type_name = "float complex";
+        // if (x.m_type == 8) type_name = "double complex";
```

Comment:
Why this is commented out?

Reply:
I got the idea from here: https://github.com/lcompilers/lpython/blob/main/src/libasr/codegen/asr_to_c.cpp#L226

but m_type is a pointer: `ttype_t* m_type`

do you have any idea how to deal with that?

another question, I copied it from another function: `last_expr_precedence = 2;`
is that correct? what is it used for?

Reply:
I tried to debug it here but I didn't find any useful information here:

```
$14 = (const LFortran::ASR::ComplexConstant_t &) @0x7ffff784c9a8: {static class_type = LFortran::ASR::ComplexConstant, base = {
    static class_type = LFortran::ASR::expr, base = {type = LFortran::ASR::expr, loc = {first = 6108, last = 6108}}, 
    type = LFortran::ASR::ComplexConstant}, m_re = 0, m_im = 0, m_type = 0x7ffff784c958}
```

any idea about how I should proceed here? thanks

Reply:
@czgdp1807 any comment about that?

Reply:
I would uncomment the line as `m_kind` signifies the number of bytes a constant/type will occupy. So for `8` kind we should `double complex`.

Reply:
yep, but as you can see here: https://github.com/lcompilers/lpython/pull/1108#discussion_r967696075

```
$14 = (const LFortran::ASR::ComplexConstant_t &) @0x7ffff784c9a8: {static class_type = LFortran::ASR::ComplexConstant, base = {
    static class_type = LFortran::ASR::expr, base = {type = LFortran::ASR::expr, loc = {first = 6108, last = 6108}}, 
    type = LFortran::ASR::ComplexConstant}, m_re = 0, m_im = 0, m_type = 0x7ffff784c958}
```
it doesn't have m_kind there. the question is, is there any way to access this information from `ASR::ComplexConstant_t`?

Reply:
m_kind is in the type of expression. Fetch the type, down cast it to ASR::Complex and then you can obtain the m_kind.